sstables/CMakeLists.txt

```diff
@@ -17,6 +17,9 @@ target_sources(sstables
     sstable_set.cc
     sstables_manager.cc
     sstable_version.cc
+    storage.cc
+    s3_storage.cc
+    filesystem_storage.cc
```

Comment:
I think you should just co-locate both in `storage.cc`, we try to avoid tiny `.cc` files as each `.cc` file has a considerable constant overhead in the amount headers that need to be processed to compile it.

Reply:
also, i think under most circumstances, we do incremental compilation, in that case, a smaller compilation unit with less dependencies is desirable. am i right?

Reply:
There is a cut-off line, above which a source file is too large and splitting it is a net improvement in compile times. Below that, the overhead dominates. In other words, I think a 100 line file compiles just as fast as a 10 line file, provided they use largely the same includes.
Note that many small source files also has a cost in linking time. Which is paid in every single incremental build (unless you compile just a `.o`).

We never measured what this is, but in practice we avoid source files which are less than a couple of hundreds of lines long.

Reply:
okay. i really wanted to focus on a certain class. probably one day, we don't need to worry too much about the compilation speed.

will move the implementations into `storage.cc` tomorrow.